### DISCUSSION TITLE
Why does this query return "null" when take > 999 (is this a dataloader limit?)

### DISCUSSION THREAD

**Discussion by HendrikJan** [prisma/prisma#13365] [Q&A] [answered]
I've got the following function:

```javascript
export default async (req, res) => {
  // Get 1000 vacancies
  const vacancies = await prisma.vacancy.findMany({
    take: 1000,
  })

  // Get the ids of the vacancies
  const ids = vacancies.map(vacancy => vacancy.id)

  // Get the staffMembers of the vacancies in one go
  const promises = [] as Promise<any>[]
  for (let i = 0; i < ids.length; i++) {
    const id = ids[i]
    promises.push(prisma.vacancy.findUnique({
      where: { id }
    }).StaffMember())
  }

  // Return the staffMembers
  const results = await Promise.all(promises)

  return results
}
```

The result from the code above is this:
```json
[
  null,
  null,
  null,
  null,
  null,
  <995 more times null>
]
```

If I change `1000` to `999` in the code above, I get:
```json
[
  {
    "id": 323,
    "type": "staff_member",
    "year_complete": false,
    "firstname": "John",
    "prefix": "",
    "surname": "Doe",
    "titlebefore": "dr.",
    "titleafter": "",
    "birthdate": "2017-01-01T00:00:00.000Z",
    "contracttype": "permanent",
    "contractend": null,
    "remarks": "some remark about this staff member",
    "notes": "",
    "created_at": "2017-01-30T15:30:54.000Z",
    "updated_at": "2017-02-22T09:22:28.000Z",
    "academic_year_id": 1,
    "account_id": 308,
    "chair_id": 28,
    "staff_member_title_id": null
  },
  <999 more times staffmember or null>
]
```

Why are all results set to `null` when `take > 999`?
It's not that record nr 1000 is a problem, because I can set `skip: 10, take: 999` and still get staff members.

I tried to reproduce the problem in a new repository, but could not get this problem to appear.

**Comment by nurul3101** [comment]
Hey @HendrikJan 👋 
What's your `@prisma/client` version?
Is this in a public repository or could you help in reproducing it by providing a sample repo?

**Reply by HendrikJan** [reply]
Hi @nurul3101 ,
I'm sorry I didn't report back on this discussion.

The problem turns out to be caused by issue #12338. (Setting `QUERY_BATCH_SIZE: 999` works as a workaround).

I've added a reproduction to the discussion in that issue.

**Comment by HendrikJan** [comment, marked as answer]
This is caused bij an issue when using Mariadb and an IN-clause with more than 999 IDs.
The issue can be found here:  #12338

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
